packages/webapp/src/components/RouterTab/Tab.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { ReactNode, CSSProperties } from 'react';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+
+export enum VARIANTS {
+  PILL = 'pill',
+  UNDERLINE = 'underline',
+}
+
+export type BaseTab = {
+  label: string;
```

Comment:
I noticed in the designs sometimes we have something else in the tab other than the label (e.g. in the Tasks tab there's a counter next to it), not a blocker at all but should we allow passing a React element here on top of a string to account for that scenario?

Reply:
I originally considered making `label` a React element as well, but I noticed that in both `RouterTab` and `StateTab`, the label is currently used as part of the `id` (`id={tab.label + index}`). To avoid potentially breaking this, I kept the `label` as a string and added a `format` function (on the next line) to include additional elements like counters or icons. (You can see an example of this in the `FormattedLabel` story.)

In such situation, I'm not really sure if I should:
1. take the time to make sure if it's safe to change `id` and make the `label` React element.
2. extend the feature without making the change to what we have.

Any guidance on which direction would be best?

Reply:
For some reason I totally missed the formatted label example -- that's perfect as is! It's even better than making label more flexible, it gives the option to customize the display of the tab without altering the label itself. Love that solution!